description: fix mismatched types on arm architectures (-Wincompatible-pointer-types)
author: Michael Gilbert <mgilbert@debian.org>

--- a/programs/winedbg/be_arm.c
+++ b/programs/winedbg/be_arm.c
@@ -1555,12 +1555,12 @@ void be_arm_disasm_one_insn(ADDRESS64 *a
     int matched = 0;
 
     char tmp[64];
-    DWORD_PTR* pval;
+    union { DWORD dvalue; struct dbg_lvalue lvalue; } values;
 
-    if (!memory_get_register(CV_ARM_CPSR, &pval, tmp, sizeof(tmp)))
+    if (!memory_get_register(CV_ARM_CPSR, &values.lvalue, tmp, sizeof(tmp)))
         dbg_printf("\n\tmemory_get_register failed: %s", tmp);
     else
-        db_disasm_thumb = (*pval & 0x20) != 0;
+        db_disasm_thumb = (values.dvalue & 0x20) != 0;
 
     db_display = display;
 
--- a/include/winber.h
+++ b/include/winber.h
@@ -22,10 +22,6 @@
 #define LBER_ERROR   (~0L)
 #define LBER_DEFAULT (~0L)
 
-typedef int ber_int_t;
-typedef unsigned int ber_tag_t;
-typedef unsigned int ber_len_t;
-
 void CDECL ber_free( BerElement *, int );
 void CDECL ber_bvfree( BERVAL * );
 
